CMakeLists.txt

```diff
@@ -9,6 +9,23 @@ cmake_minimum_required(VERSION 3.9)
 project(Cabana LANGUAGES CXX)
 set(PROJECT_VERSION "0.3.0-dev")
 
+# Download and unpack googletest at configure time
+configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
+execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download" )
+execute_process(COMMAND "${CMAKE_COMMAND}" --build .
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download" )
```

Comment:
This seems to be a hackery to make is work with older versions of cmake, I think we should require cmake-3.11 and use [FetchContent](https://cmake.org/cmake/help/v3.11/module/FetchContent.html) instead:
```
FetchContent_Declare(
  googletest
  GIT_REPOSITORY https://github.com/google/googletest.git
  GIT_TAG        release-1.10.0
)

FetchContent_GetProperties(googletest)
if(NOT googletest_POPULATED)
  FetchContent_Populate(googletest)
  add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR})
endif()
```

Reply:
I'm fine with that - I just basically copy/pasted what was in the GTest manual

Reply:
so no `CMakeLists.txt.in` then?

Reply:
Yeah.

Reply:
My code snippet above is actually taken for the CMake manual ;-)

Reply:
ok - fixed